### DISCUSSION TITLE
Error while building MMDeployCSharp in VS Code

### DISCUSSION THREAD

**Discussion by pd2871** [open-mmlab/mmdeploy#1674] [Q&A]
@lvhan028  might probably help me on this, I downloaded all the folders from the CSharp demo of MMDeploy and tried to build the solution, but it is throwing error which says that "Failed to retrieve information about  MMDeployCSharp from remote source"  which means the library is not found in the nuget package, what can the possible reason for this? Any help will be highly appreciated, Thanks

**Comment by lvhan028** [comment]
@irexyc could you help deal with it?

**Comment by pd2871** [comment]
@irexyc I built the MMDeploy nuget package using this [link](https://github.com/open-mmlab/mmdeploy/tree/master/csrc/mmdeploy/apis/csharp)

The when I try to build the `demo.sln` from this [link](https://github.com/open-mmlab/mmdeploy/tree/master/demo/csharp),

it is throwing `The remote name could not be resolved 'packagesource'`, 

Any help will be appreciated.
```
Rebuild started...
Retrying 'FindPackagesByIdAsyncCore' for source 'https://packagesource/FindPackagesById()?id='MMDeployCSharp'&semVerLevel=2.0.0'.
An error occurred while sending the request.
  The remote name could not be resolved: 'packagesource'
Retrying 'FindPackagesByIdAsyncCore' for source 'https://packagesource/FindPackagesById()?id='OpenCvSharp4.runtime.win'&semVerLevel=2.0.0'.
An error occurred while sending the request.
  The remote name could not be resolved: 'packagesource'
Retrying 'FindPackagesByIdAsyncCore' for source 'https://packagesource/FindPackagesById()?id='OpenCvSharp4'&semVerLevel=2.0.0'.
An error occurred while sending the request.
  The remote name could not be resolved: 'packagesource'
Retrying 'FindPackagesByIdAsyncCore' for source 'https://packagesource/FindPackagesById()?id='OpenCvSharp4'&semVerLevel=2.0.0'.
An error occurred while sending the request.
  The remote name could not be resolved: 'packagesource'
Retrying 'FindPackagesByIdAsyncCore' for source 'https://packagesource/FindPackagesById()?id='MMDeployCSharp'&semVerLevel=2.0.0'.
An error occurred while sending the request.
  The remote name could not be resolved: 'packagesource'
Retrying 'FindPackagesByIdAsyncCore' for source 'https://packagesource/FindPackagesById()?id='System.Memory'&semVerLevel=2.0.0'.
An error occurred while sending the request.
```

**Comment by irexyc** [comment]
Hi @pd2871, I have no clue why you can't install `OpenCvSharp4`, `System.Memory` and `OpenCvSharp4.runtime.win` as they are in nuget source https://www.nuget.org/packages/OpenCvSharp4#versions-body-tab. To install MMDeployCSharp you need to add a local package sources.
<img width="691" alt="image" src="https://user-images.githubusercontent.com/16019484/217218906-21ae8838-ab6b-4a57-888d-1c317e9a67f1.png">


Can you install them manually?
<img width="1320" alt="image" src="https://user-images.githubusercontent.com/16019484/217218430-67b20747-23a8-4bed-bbd1-371b8aa8b5ae.png">

**Comment by pd2871** [comment]
Hi @irexyc , Thanks for the images, the problem with this error was, it was using `https://packagesource` as a default URL to search the packages instead of `nuget.org`, your first image gave me the clue.
So, when I unchecked that 1st option from package sources of `Nuget Package Manager`, it seemed to find the packages from `nuget.org` as a default, and the error was resolved `(Image shown below)`. Thank you once again,

Also, it would be great if you can update `Readme.md` of both build and demo of csharp, as some of the `links/paths` given in `Readme.md` are broken as of now. 
And also, it would be great if you can add the link of `demo page` of csharp at the end of the `API build page` of csharp in `Readme.md`, so that users can directly go to demo page right after building the SDK.  Would also be great to include this info in `Readme.md` of main page of MMDeploy so that new users can easily search the C# API building/demo process from the main page itself.

![Screenshot (2800)](https://user-images.githubusercontent.com/59336638/217300313-e1a6862c-91ea-401d-9ff2-786570275f69.png)
